File: rle-iterator/rle-iterator.py
class RLEIterator:

    def __init__(self, encoding: List[int]):
        self.encoding = encoding
        self.i = 0
        self.n = len(self.encoding)

    def next(self, n: int) -> int:
        while self.i < self.n and n > self.encoding[self.i]:
            n -= self.encoding[self.i]
            self.i += 2
        if self.i >= self.n:
            return -1
        self.encoding[self.i] -= n
        return self.encoding[self.i + 1]


# Your RLEIterator object will be instantiated and called as such:
# obj = RLEIterator(encoding)
# param_1 = obj.next(n)

File: rle-iterator/rle-iterator.c



typedef struct {
    int* encoding;
    int size;
    int i;
} RLEIterator;


RLEIterator* rLEIteratorCreate(int* encoding, int encodingSize) {
    RLEIterator* ret = malloc(sizeof(RLEIterator));
    ret->encoding = encoding;
    ret->size = encodingSize;
    ret->i = 0;
    return ret;
}

int rLEIteratorNext(RLEIterator* obj, int n) {
    while (obj->i < obj->size && n > obj->encoding[obj->i]){
        n -= obj->encoding[obj->i];
        obj->i += 2;
    }
    if (obj->i >= obj->size){
        return -1;
    }
    obj->encoding[obj->i] -= n;
    return obj->encoding[(obj->i + 1)];
}

void rLEIteratorFree(RLEIterator* obj) {
    free(obj);
}

/**
 * Your RLEIterator struct will be instantiated and called as such:
 * RLEIterator* obj = rLEIteratorCreate(encoding, encodingSize);
 * int param_1 = rLEIteratorNext(obj, n);
 
 * rLEIteratorFree(obj);
*/